src/components/InpuField/InputField.tsx

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import IconButton from "@mui/material/IconButton";
+import SearchIcon from "@mui/icons-material/Search";
+import CloseIcon from "@mui/icons-material/Close";
+import { InputFieldWrapper, StyledInputField } from "./styles";
+
+interface CustomSearchInputProps {
+  value: string;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  onSubmit: () => void;
+  onClear: () => void;
+  placeholder?: string;
+  disabled?: boolean;
+  startIcon?: string;
+  endIcon?: string;
+  sx?: object;
+}
+
+const InputField = ({
+  value,
+  onChange,
+  onSubmit,
+  onClear,
+  placeholder = "Search",
+  disabled = false,
+  startIcon,
+  endIcon,
+  sx = {},
+}: CustomSearchInputProps) => {
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === "Enter") {
+      onSubmit();
+    }
+  };
+
+  return (
+    <InputFieldWrapper $disabled={disabled}>
+      <IconButton onClick={onSubmit} disabled={disabled}>
```

Comment:
You need to choose if the IconButton is always a submit button and then you don't need the `startIcon`, or you can custom the `startIcon` and then maybe you don't want it to be a submit button.
Another option is just change the onSubmit to onStartIconClick maybe

Reply:
Changed to onStartIconClick, is this what you meant?